src/Core/Resolvers/MsSqlQueryExecutor.cs

```diff
@@ -83,6 +85,49 @@ public MsSqlQueryExecutor(
             ConfigureMsSqlQueryEecutor();
         }
 
+        /// <summary>
+        /// Creates a SQLConnection to the data source of given name. This method also adds an event handler to
+        /// the connection's InfoMessage to extract the statement ID from the request and add it to httpcontext.
+        /// </summary>
+        /// <param name="dataSourceName">The name of the data source.</param>
+        /// <returns>The SQLConnection</returns>
+        /// <exception cref="DataApiBuilderException">Exception thrown if datasource is not found.</exception>
+        public override SqlConnection CreateConnection(string dataSourceName)
+        {
+            if (!ConnectionStringBuilders.ContainsKey(dataSourceName))
+            {
+                throw new DataApiBuilderException("Query execution failed. Could not find datasource to execute query against", HttpStatusCode.BadRequest, DataApiBuilderException.SubStatusCodes.DataSourceNotFound);
+            }
+
+            SqlConnection conn = new()
+            {
+                ConnectionString = ConnectionStringBuilders[dataSourceName].ConnectionString,
+            };
+
+            // If connection is a SQLConnection and it is not null, we can extract the info message
```

Comment:
where do we check if conn is not null?

Reply:
Good point! That was an outdated comment. We create the connection right before this line, so the connection should not be null

Reply:
What if the connection creation fails? Do we have enough safeguards against that to handle gracefully?

Reply:
Added some more checks to:
- Throw an exception (with a helpful error message) in case the connection is null
- Log an error but continue the execution in case there was an exception thrown in the logic that adds the statement ID to httpcontext